packages/plugins/response-cache/src/index.ts

```diff
@@ -129,61 +126,69 @@ export function useResponseCache(options: UseResponseCacheParameter): Plugin {
         }),
       );
     },
-    async onRequest({ request, fetchAPI, endResponse }) {
-      if (enabled(request)) {
+    onRequest({ request, serverContext, fetchAPI, endResponse }): PromiseOrValue<void> {
+      if (enabled(request, serverContext as TContext)) {
         const operationId = request.headers.get('If-None-Match');
         if (operationId) {
-          const cachedResponse = await cache.get(operationId);
-          if (cachedResponse) {
-            const lastModifiedFromClient = request.headers.get('If-Modified-Since');
-            const lastModifiedFromCache =
-              cachedResponse.extensions?.http?.headers?.['Last-Modified'];
-            if (
-              // This should be in the extensions already but we check it here to make sure
-              lastModifiedFromCache != null &&
-              // If the client doesn't send If-Modified-Since header, we assume the cache is valid
-              (lastModifiedFromClient == null ||
-                new Date(lastModifiedFromClient).getTime() >=
-                  new Date(lastModifiedFromCache).getTime())
-            ) {
-              const okResponse = new fetchAPI.Response(null, {
-                status: 304,
-                headers: {
-                  ETag: operationId,
-                },
-              });
-              endResponse(okResponse);
+          return mapMaybePromise(cache.get(operationId), cachedResponse => {
```

Comment:
What is the point of adding this `mapMaybePromise` thing here? Does it improve the performance measurably or is it just in case?

Reply:
Avoiding async/await(unnecessary promises) always increases the performance. However, Yoga should already need a refactor to avoid async/await.

Reply:
I think Laurin is aluding to the fact that cache.get always returns a promise, it'll never not be a promise.